src/lib/engine/space.js

```diff
@@ -12,10 +12,11 @@ export function get(collection, id){
   return collection.find(i => (i.id === id))
 }
 
-export function subset(graph, spaceId){
+export function subset(graph, spaceId, withInputs = false){
   if (spaceId){
     const metrics = graph.metrics.filter(m => m.space === spaceId)
-    const guesstimates = _.flatten(metrics.map(m => _metric.guesstimates(m, graph)))
+    const rawGuesstimates = _.flatten(metrics.map(m => _metric.guesstimates(m, graph)))
+    const guesstimates = withInputs ? rawGuesstimates.map(_guesstimate.inputsFromExpressions(metrics)) : rawGuesstimates
```

Comment:
I believe this runs _guesstimate.inputsFromExpressions(metrics) n times.  Can you save it above as a ``const``?  Speed aside, that would make it a tad bit more understandable.